src/witan/send/send.clj

```diff
@@ -342,12 +345,61 @@
   (> current-year-in-loop projection-end-year))
 
 ;;Post-loop functions
+(defn quantile ;; from criterium
+               ;; https://crossclj.info/ns/criterium/0.4.4/criterium.stats.html#_quantile
+  "Calculate the quantile of a sorted data set
+   References: http://en.wikipedia.org/wiki/Quantile"
+  [quantile data]
+  (let [n (dec (count data))
+        interp (fn [x]
+                 (let [f (Math/floor x)
+                       i (long f)
+                       p (- x f)]
+                   (+ (* p (nth data (inc i))) (* (- 1.0 p) (nth data i)))))]
+    (interp (* quantile n))))
+
+(defn quantile* ;; from https://gist.github.com/scottdw/2960070
+  ([p vs]
+   (let [svs (sort vs)]
+     (quantile* p (count vs) svs (first svs) (last svs))))
+  ([p c svs mn mx]
+   (let [pic (* p (inc c))
+         k (int pic)
+         d (- pic k)
+         ndk (if (zero? k) mn (nth svs (dec k)))]
+     (cond
+       (zero? k) mn
+       (= c (dec k)) mx
+       (= c k) mx
+       :else (+ ndk (* d (- (nth svs k) ndk)))))))
+
+(defn confidence-interval-and-mean
+  "Takes in a dataset of send projections and outputs average and confidence interval boundaries."
+  [grouped-projection]
+  (->> grouped-projection
+       (pmap (fn [[k v]] (let [state (:state k)
+                               sims-counts (frequencies (ds/column v :sim-num))
+                               mean (/ (apply + (vals sims-counts)) (count sims-counts))
+                               sorted-counts (sort (vals sims-counts))
+                               low-ci (quantile* 0.025 sorted-counts)
+                               high-ci (quantile* 0.975 sorted-counts)
+                               split-keyword (fn [kw] (clojure.string/split (name kw) #"-"))
```

Comment:
I encouraged this for the purposes of speed. In fact, I think we might see a few more corners being cut as we trade principles for efficiency!